### ISSUE TITLE
Convert from reStructuredText: unable to recognize embedded URIs

### ISSUE THREAD

**Issue by ivangalkin** [jgm/pandoc#4912]
Hi All,

pandoc (tested versions 2.2.3.2 and 2.3) is unable to parse embedded URIs of RST input files in some cases. For perfectly fine RST input (see the [official reference](http://docutils.sourceforge.net/docs/user/rst/quickref.html#external-hyperlink-targets)) and the valid hyperlinks like...

```reStructuredText
`Forwarding references
<http://www.open-std.org/jtc1/sc22/wg21/docs/papers/2014/n4164.pdf>`_
```

... pandoc prints a warning... 

```
[WARNING] Reference not found for 'forwarding references <http://www.open-std.org/jtc1/sc22/wg21/docs/papers/2014/n4164.pdf>' at line 16 column 70
```

.. and produces a wrong output. E.g. HTML

```HTML
<a href="">Forwarding references &lt;http://www.open-std.org/jtc1/sc22/wg21/docs/papers/2014/n4164.pdf&gt;</a>
```

How to reproduce:

I observed this error as I tried to convert the clang-tidy documentation from reStructuredText to HTML in the batch manner (clang online documentation is generated by means of sphinx).

1. cd /tmp && svn co http://llvm.org/svn/llvm-project/clang-tools-extra/trunk
2. cd /tmp/trunk/docs/clang-tidy/checks
3. create and execute the following bash script
```BASH
FILES=*.rst
for f in $FILES
do
  filename="${f%.*}"
  (>&2 echo "Converting $f to $filename.html")
  `pandoc $f -f rst -t html -o $filename.html`
done
```

4. this will produce the following warnings

Alternatively you just can call pandoc manually for each single file from the list below. All affected files are also available under http://llvm.org/svn/llvm-project/clang-tools-extra/trunk/docs/clang-tidy/checks/ .

Thank you!

```
Converting abseil-no-namespace.rst to abseil-no-namespace.html
[WARNING] Reference not found for 'the full abseil compatibility guidelines <https:// abseil.io/about/compatibility>' at line 21 column 33
Converting android-comparison-in-temp-failure-retry.rst to android-comparison-in-temp-failure-retry.html
[WARNING] Reference not found for 'a convenience macro <https://www.gnu.org/software/libc/manual/html_node/interrupted-primitives.html>' at line 12 column 83
Converting bugprone-move-forwarding-reference.rst to bugprone-move-forwarding-reference.html
[WARNING] Reference not found for 'forwarding references <http://www.open-std.org/jtc1/sc22/wg21/docs/papers/2014/n4164.pdf>' at line 16 column 70
Converting cert-dcl21-cpp.rst to cert-dcl21-cpp.html
[WARNING] Reference not found for 'dcl21-cpp. overloaded postfix increment and decrement operators should return a const object <https://www.securecoding.cert.org/confluence/display/cplusplus/dcl21-cpp.+overloaded+postfix+increment+and+decrement+operators+should+return+a+const+object>' at line 12 column 160
Converting cert-dcl50-cpp.rst to cert-dcl50-cpp.html
[WARNING] Reference not found for 'dcl50-cpp. do not define a c-style variadic function <https://www.securecoding.cert.org/confluence/display/cplusplus/dcl50-cpp.+do+not+define+a+c-style+variadic+function>' at line 11 column 120
Converting cert-dcl58-cpp.rst to cert-dcl58-cpp.html
[WARNING] Reference not found for 'dcl58-cpp. do not modify the standard namespaces <https://www.securecoding.cert.org/confluence/display/cplusplus/dcl58-cpp.+do+not+modify+the+standard+namespaces>' at line 21 column 116
Converting cert-env33-c.rst to cert-env33-c.html
[WARNING] Reference not found for 'env33-c. do not call system() <https://www.securecoding.cert.org/confluence/pages/viewpage.action?pageid=2130132>' at line 13 column 86
Converting cert-err34-c.rst to cert-err34-c.html
[WARNING] Reference not found for 'err34-c. detect errors when converting a string to a number <https://www.securecoding.cert.org/confluence/display/c/err34-c.+detect+errors+when+converting+a+string+to+a+number>' at line 28 column 119
Converting cert-err52-cpp.rst to cert-err52-cpp.html
[WARNING] Reference not found for 'err52-cpp. do not use setjmp() or longjmp() <https://www.securecoding.cert.org/confluence/pages/viewpage.action?pageid=1834>' at line 10 column 83
Converting cert-err58-cpp.rst to cert-err58-cpp.html
[WARNING] Reference not found for 'err58-cpp. handle all exceptions thrown before main() begins executing <https://www.securecoding.cert.org/confluence/display/cplusplus/err58-cpp.+handle+all+exceptions+thrown+before+main%28%29+begins+executing>' at line 11 column 142
Converting cert-err60-cpp.rst to cert-err60-cpp.html
[WARNING] Reference not found for 'err60-cpp. exception objects must be nothrow copy constructible <https://www.securecoding.cert.org/confluence/display/cplusplus/err60-cpp.+exception+objects+must+be+nothrow+copy+constructible>' at line 11 column 131
Converting cert-flp30-c.rst to cert-flp30-c.html
[WARNING] Reference not found for 'flp30-c. do not use floating-point variables as loop counters <https://www.securecoding.cert.org/confluence/display/c/flp30-c.+do+not+use+floating-point+variables+as+loop+counters>' at line 11 column 121
Converting cert-msc51-cpp.rst to cert-msc51-cpp.html
[WARNING] Reference not found for 'msc51-cpp. ensure your random number generator is properly seeded <https://wiki.sei.cmu.edu/confluence/display/cplusplus/msc51-cpp.+ensure+your+random+number+generator+is+properly+seeded>' at line 13 column 124
[WARNING] Reference not found for 'msc32-c. properly seed pseudorandom number generators <https://wiki.sei.cmu.edu/confluence/display/c/msc32-c.+properly+seed+pseudorandom+number+generators>' at line 15 column 104
Converting hicpp-no-assembler.rst to hicpp-no-assembler.html
[WARNING] Reference not found for 'high intergrity c++ coding standard <http://www.codingstandard.com/section/7-5-the-asm-declaration/>' at line 9 column 67
Converting misc-new-delete-overloads.rst to misc-new-delete-overloads.html
[WARNING] Reference not found for 'dcl54-cpp. overload allocation and deallocation functions as a pair in the same scope <https://www.securecoding.cert.org/confluence/display/cplusplus/dcl54-cpp.+overload+allocation+and+deallocation+functions+as+a+pair+in+the+same+scope>' at line 19 column 153
Converting misc-non-copyable-objects.rst to misc-non-copyable-objects.html
[WARNING] Reference not found for 'fio38-c. do not copy a file object <https://www.securecoding.cert.org/confluence/display/c/fio38-c.+do+not+copy+a+file+object>' at line 13 column 94
Converting misc-throw-by-value-catch-by-reference.rst to misc-throw-by-value-catch-by-reference.html
[WARNING] Reference not found for 'throw anonymous temporaries <https://www.securecoding.cert.org/confluence/display/cplusplus/err09-cpp.+throw+anonymous+temporaries>' at chunk line 4 column 106
Converting modernize-pass-by-value.rst to modernize-pass-by-value.html
[WARNING] Reference not found for 'want speed? pass by value' at chunk line 3 column 87
Converting performance-implicit-cast-in-loop.rst to performance-implicit-cast-in-loop.html
[WARNING] Reference not found for 'performance-implicit-conversion-in-loop <performance-implicit-conversion-in-loop.html>' at line 11 column 49
Converting readability-else-after-return.rst to readability-else-after-return.html
[WARNING] Reference not found for 'llvm coding standards recommendation <http://llvm.org/docs/codingstandards.html#don-t-use-else-after-a-return>' at line 64 column 76
Converting readability-implicit-bool-cast.rst to readability-implicit-bool-cast.html
[WARNING] Reference not found for 'readability-implicit-bool-conversion <readability-implicit-bool-conversion.html>' at line 11 column 46

```

**Comment by jgm** [comment]
Interesting:

```
% pandoc -f rst
`Foo
<http://example.com>`_
[WARNING] Reference not found for 'foo <http://example.com>' at line 2 column 23
<p><a href="">Foo &lt;http://example.com&gt;</a></p>
```
but
```
% pandoc -f rst
`Foo <http://example.com>`_
<p><a href="http://example.com">Foo</a></p>
```
so it's getting tripped up on the newline.

**Comment by jgm** [comment]
@danse was there a reason `endline` was excluded from `inlineContent`?  If not, then perhaps these issues can be fixed just by adding it.  If so, then this will need more thought.

**Comment by danse** [comment]
hey, thanks for pointing my attention to this issue, i'll look up the details

**Comment by danse** [comment]
thanks to you for the detailed report :)

**Comment by jgm** [comment]
Closed by #4914
